torch/distributed/checkpoint/state_dict_saver.py

```diff
@@ -23,18 +27,22 @@
 from torch.distributed.checkpoint._storage_utils import _storage_setup
 from torch.distributed.checkpoint.default_planner import DefaultSavePlanner
 from torch.distributed.checkpoint.logger import _dcp_method_logger
-from torch.distributed.checkpoint.metadata import Metadata, STATE_DICT_TYPE
+from torch.distributed.checkpoint.metadata import Metadata
 from torch.distributed.checkpoint.planner import SavePlan, SavePlanner
 from torch.distributed.checkpoint.staging import AsyncStager
+from torch.distributed.checkpoint.staging import AsyncStager
 from torch.distributed.checkpoint.stateful import Stateful
 from torch.distributed.checkpoint.storage import StorageWriter
 from torch.distributed.distributed_c10d import _get_default_group
+from typing_extensions import deprecated
 
 from .utils import _api_bc_check, _DistWrapper, _profile
 
 
 __all__ = ["save_state_dict", "save", "async_save", "AsyncCheckpointerType"]
 
+CACHED_STATE_DICT: Optional[STATE_DICT_TYPE] = None
```

Comment:
We are not going to expose this variable to users, so we need to make it a private variable.